Created traces beta announcement (#17192)
PostHog/posthog.com: contents/blog/traces-beta.mdx

---

> title: 'The clues were there all along. Traces (beta) is live.'

Reviewer: ```suggestion
title: 'The clues were there all along. Traces is now in beta!'
```

> date: 2026-06-02

Reviewer: ```suggestion
date: 2026-06-05
```

rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
author:
    - sara-miteva
featuredImage: >-
    https://res.cloudinary.com/dmukukwp6/image/upload/Untitled_January_9_2026_15_26_39_copy_2_7_1_f2bff5a46d.png
featuredImageType: full
category: PostHog news
tags:
    - Product updates
    - Tracing
seo:
    metaTitle: 'Traces is now in beta'
    metaDescription: 'PostHog Traces is now in beta. Point your existing OpenTelemetry exporter at PostHog and get distributed tracing alongside your replays, errors, logs, and user data.'
---

> Traces is now in beta. If you've been using [PostHog Logs](/logs), you probably sensed this was coming.

Reviewer: ```suggestion
Traces is now in beta. If you've been using [PostHog Logs](/logs), you probably saw this coming.
```

> PostHog starts with your user data and provides you with all the tools you need to build successful products. Debugging is one of the processes you definitely need to keep your products working as expected.

Reviewer: ```suggestion
PostHog starts with your user data and provides you with all the tools you need to build successful products, so debugging is an obvious must-have to keep your products working as expected.
```

> So far, we've had [Error tracking](/error-tracking), [Session replay](/session-replay), and [Logs](/logs) in the PostHog debugging stack, and Traces was a logical addition that will give you more of the context you need to resolve issues quickly.

Reviewer: ```suggestion
We have [Error tracking](/error-tracking), [Session replay](/session-replay), and [Logs](/logs) in the PostHog debugging stack, so Traces was a logical addition to give you more of the context you need to resolve issues quickly.
```

## Why we built it this way (spoiler alert: not just another tracing tool)

Every observability tool we've used makes the same assumption: a trace is something an SRE looks at on a dashboard. Spans, services, latency, hosts. The user who hit the bug is somewhere else, in your analytics, your session replay, your customer support tool. This means you have to juggle all these tools to find the user context.

> PostHog flips that (or should I use more SaaS-friendly talk and say we're DISRUPTING the market). We built observability on top of analytics, not the other way around. So when a span lands in PostHog, it lands in the same project as your replays, your errors, your logs, and the user who triggered it. Same identity, same session, same event stream. Every span has a face.

Reviewer: ```suggestion
We built observability _on top of_ analytics instead. So when a span lands in PostHog, it lands in the same project as your replays, your errors, your logs, and the user who triggered it. Same identity, same session, same event stream. Every span has a face.
```

> That shared foundation is live in the beta. The workflows that pay it off: click a span and watch the replay, pivot from a slow request to the customer it cost you, are what's next on the roadmap, not in the box today. We wanted to ship the honest version first.

Reviewer: This seems very AI-y, I'd delete unless you can make it much clearer!

## What's in the beta

Point your existing OpenTelemetry exporter at PostHog's OTLP/HTTP endpoint, drop in a bearer token, and you're sending traces. No new SDK, no agent on the box, no rewrite of the instrumentation you already have.

```bash
OTEL_EXPORTER_OTLP_TRACES_ENDPOINT="<your-posthog-host>/i/v1/traces"
OTEL_EXPORTER_OTLP_TRACES_HEADERS="Authorization=Bearer <your-project-api-key>"
OTEL_SERVICE_NAME="my-app"
```

> That's the config change.

Reviewer: ```suggestion
```

Inside PostHog, you get a volume sparkline for traffic at a glance, a filterable table of root spans (slice by service, name, kind, status, duration, trace or span ID, or any attribute on the span), and a flamegraph for any trace you click into. All of it lives on the same ClickHouse fleet as your PostHog Logs, so the same query layer runs both.

It's the boring, working version of distributed tracing, already sitting next to the rest of your PostHog data.

## Debugging with PostHog

Say a user complains your checkout is slow. Because their `user_id` is on both your OTel spans and your PostHog events, you can pull up every span they triggered in the last hour and find the four-second wait on a downstream payment provider. From there, opening their session replay in another tab shows exactly what they were staring at while the request hung: same person, same session, all in the project you're already in.

> That's a workflow you can run on day one. What we haven't built yet is the connective tissue that makes it feel native: clicking a slow span and having the replay open beside it, or clicking an error and watching the trace it caused appear in line. The shared identity layer is what makes those jumps possible, and it's already there. The UI to skip the tab-juggling is next on the roadmap.

Reviewer: I'd probably delete this, extra words not needed

> Once those links are wired, this is where PostHogAI starts to earn its keep, answering something like *"show me the slowest spans this week for users on the Pro plan who hit feature flag X"* with one prompt instead of cobbling together four queries across four tools.

Reviewer: ```suggestion
Once those links are wired, you can use PostHog AI to answer questions like *"show me the slowest spans this week for users on the Pro plan who hit feature flag X"* with one prompt instead of cobbling together four queries across four tools.
```

---

> title: 'The clues were there all along. Tracing is now in beta!'

Reviewer 2: ```suggestion
title: 'The clues were there all along. Tracing is now in beta'
```

> date: 2026-06-02

Reviewer 2: ```suggestion
date: 2026-07-16
```

rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
author:
    - sara-miteva
featuredImage: >-
    https://res.cloudinary.com/dmukukwp6/image/upload/Untitled_January_9_2026_15_26_39_copy_2_7_1_f2bff5a46d.png
featuredImageType: full
category: PostHog news
tags:
    - Product updates
    - Tracing
seo:

>     metaTitle: 'Traces is now in beta'

Reviewer 3: My feedback is mostly that we are standardizing on calling the product `Tracing` or `Distributed Tracing` over just traces (since AI observability has traces already). So we should change that, in this article.

    metaDescription: 'PostHog distributed tracing is now in beta. Point your existing OpenTelemetry exporter at PostHog and get distributed tracing alongside your replays, errors, logs, and user data.'
---

[Distributed tracing](/docs/distributed-tracing) is now in beta. If you've been using [PostHog Logs](/logs), you probably saw this coming.

PostHog starts with your user data and provides you with all the tools you need to build successful products, so debugging is an obvious must-have to keep your products working as expected.

> We have [Error tracking](/error-tracking), [Session replay](/session-replay), and [Logs](/logs) in the PostHog debugging stack, so tracing was a logical addition to give you more of the context you need to resolve issues quickly, and more fuel for the agents doing it for you. 

Reviewer 2: ```suggestion
We have [Error Tracking](/error-tracking), [Session Replay](/session-replay), and [Logs](/logs) in the PostHog debugging stack, so tracing was a logical addition to give you more of the context you need to resolve issues quickly, and more fuel for the agents doing it for you. 
```

> We have [Error tracking](/error-tracking), [Session replay](/session-replay), and [Logs](/logs) in the PostHog debugging stack, so tracing was a logical addition to give you more of the context you need to resolve issues quickly, and more fuel for the agents doing it for you. 

Reviewer 2: We sort of say the same thing twice here "you saw this coming" and "logical additional", could we instead use the final paragraph to say what traces is and what it's features are? I'm unfamiliar so a one sentence explainer could be helpful.

## More fuel for your agents

> Most observability tools assume a trace is something an SRE stares at on a dashboard: spans, services, latency, hosts. The user who hit the bug lives somewhere else, in your analytics, your replay, your support tool, so you end up juggling tools to work out who was actually affected. 

Reviewer 2: ```suggestion
Most observability tools assume a trace is something a site reliability engineer stares at on a dashboard: spans, services, latency, hosts. The user who hit the bug lives somewhere else, in your analytics, your replay, your support tool, so you end up juggling tools to work out who was actually affected. 
```

We built observability on top of all our existing data instead. When a span lands in PostHog, it lands in the same project as your replays, your errors, your logs, and the user who triggered it. 

That shared foundation is also what makes traces so useful to a self-driving product. Errors, replays, and funnels tell an agent that something's wrong. A trace tells it where and why: "checkout took 3.2s, and 2.8s of it was waiting on an N+1 query in the inventory service." 

![report investigate mock](https://res.cloudinary.com/dmukukwp6/image/upload/Report_Investigate_Mock_2_76290ef07e.png) 

So a scout can catch the regression, an agent can trace it to the exact line, open a PR with the fix, and drop it in your [Inbox](https://app.posthog.com/inbox). You hit merge. That's the whole job.

> ## What's in the beta

Reviewer 2: I feel like this section might be more relevant to start with than the "More fuel for your agents" one. More fuel comes after a basic overview of what it is.

Point your existing OpenTelemetry exporter at PostHog's OTLP/HTTP endpoint, drop in a bearer token, and you're sending traces. There's no need for a new SDK or rewriting the instrumentation you already have.

```bash

> OTEL_EXPORTER_OTLP_TRACES_ENDPOINT="<your-posthog-host>/i/v1/traces"

Reviewer 2: ```suggestion
OTEL_EXPORTER_OTLP_TRACES_ENDPOINT="<ph_client_api_host>/i/v1/traces"
```

> OTEL_EXPORTER_OTLP_TRACES_HEADERS="Authorization=Bearer <your-project-api-key>"

Reviewer 2: ```suggestion
OTEL_EXPORTER_OTLP_TRACES_HEADERS="Authorization=Bearer <ph_project_token>"
```

OTEL_SERVICE_NAME="my-app"
```

> Inside PostHog, you get a volume sparkline for traffic at a glance, a filterable table of root spans (slice by service, name, kind, status, duration, trace or span ID, or any attribute on the span), and a flamegraph for any trace you click into. All of it lives on the same ClickHouse fleet as your PostHog Logs, so the same query layer runs both.

Reviewer 2: ```suggestion
Inside PostHog, you get a volume sparkline for traffic at a glance, a filterable table of root spans (slice by service, name, kind, status, duration, trace or span ID, or any attribute on the span), and a flamegraph for any trace you click into. All of it lives on the same warehouse as your PostHog Logs, so the same query can run across both.
```

![traces screenshot](https://res.cloudinary.com/dmukukwp6/image/upload/Group_144143_b7a6b4aa06.png)

## Debugging with PostHog

Say a user complains your checkout is slow. Because their `user_id` is on both your OTel spans and your PostHog events, you can pull up every span they triggered in the last hour and find the four-second wait on a downstream payment provider. From there, opening their session replay in another tab shows exactly what they were staring at while the request hung: same person, same session, all in the project you're already in.

![debugging with traces and session replay](https://res.cloudinary.com/dmukukwp6/image/upload/traces_blog_illo_a355e7cfa8.jpg)

> Whether you chase it down yourself or let a scout do it, it's the same context underneath. Once those links are wired, you could also use PostHogAI to answer questions like: 

Reviewer 2: ```suggestion
Whether you chase it down yourself or let a [scout](/docs/self-driving/scouts) do it, it's the same context underneath. Once those links are wired, you could also use [PostHog AI](/ai) or our [MCP](/mcp) to answer questions like: 
```

> ```

Reviewer 2: ```suggestion
```llm
```

Show me the slowest spans this week for users on the Pro plan who hit feature flag X.
```

> [Try it with PostHogAI](https://app.posthog.com/#panel=max:Show%20me%20the%20slowest%20spans%20this%20week%20for%20users%20on%20the%20Pro%20plan%20who%20hit%20feature%20flag%20X.)

Reviewer 2: The code block already has this link
```suggestion
```

## Try it

Tracing is in beta now. The fastest way to get to it in is the wizard, which sets up self-driving and points your traces at PostHog for you:

```bash
npx @posthog/wizard@latest self-driving

> ```

Reviewer 2: We have a fancy version we can use here. 
```suggestion
<WizardCommand command="self-driving" />
```
